### PR TITLE
Bump version to v6.2.3

### LINKED ISSUE #305: There is a new release of the Conjur CLI
In particular this release will include the fix from #303

### PR DESCRIPTION
**Pull request by izgeri**
### What does this PR do?
- Bumps the version for 6.2.3
- Updates pipeline to only publish new images / gems on tags

### What ticket does this PR close?
Resolves #305 

#### Change log
- [x] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [ ] This PR includes new unit and integration tests to go with the code changes, or
- [x] The changes in this PR do not require tests

#### Documentation
- [ ] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [x] This PR does not require updating any documentation

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,9 +5,12 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
...
 
 ## [Unreleased]
+
+## [6.2.3] - 2020-12-22
 ### Fixed
-- The Conjur CLI now raises a proper error when trying to rotate a non-existing user's API key
-  ([cyberark/conjur#979](https://github.com/cyberark/conjur/issues/979))
+- The Conjur CLI now raises a proper error when trying to rotate a non-existing
+  user's API key.
+  [cyberark/conjur#979](https://github.com/cyberark/conjur/issues/979)
```
Comment (Tovli) [inline CHANGELOG.md:13]: All other changes haven't got a new line for the issue... do we care for that?

File: Jenkinsfile [hunk 2]
```diff
@@ -109,9 +109,7 @@ pipeline {
     }
 
     stage('Push standalone image to DockerHub') {
-      when {
-        branch 'master'
-      }
+      when { tag "v*" }
```
Comment (Tovli) [inline Jenkinsfile:112]: Don't we want this to be on a separated PR? it seems like a big change and might be confusing when searching through PR list as the title suggest only a version bump

File: conjur-cli.gemspec [hunk 3]
```diff
@@ -3,11 +3,11 @@ require File.expand_path('../lib/conjur/version', __FILE__)
 require "English"
 
 Gem::Specification.new do |gem|
-  gem.authors       = ["Rafal Rzepecki", "Kevin Gilpin"]
-  gem.email         = ["rafal@conjur.net", "kgilpin@conjur.net",]
+  gem.authors       = ["Conjur Maintainers"]
+  gem.email         = ["conj_maintainers@cyberark.com",]
```
Comment (Tovli) [inline conjur-cli.gemspec:7]: Didn't know we have a DL :)

File: conjur-cli.gemspec [hunk 4]
```diff
@@ -3,11 +3,11 @@ require File.expand_path('../lib/conjur/version', __FILE__)
...
   gem.summary       = %q{Conjur command line interface}
-  gem.homepage      = "https://github.com/conjurinc/cli-ruby"
-  gem.license       = 'MIT'
+  gem.homepage      = "https://github.com/cyberark/conjur-cli"
+  gem.license       = 'Apache 2.0'
```
Comment (Tovli) [inline conjur-cli.gemspec:10]: Don't we want an issue for that? to explain this change in licensing to the community?

**Review by Tovli — CHANGES_REQUESTED** [review]
Overall my concern is that the title suggest a minor change in the version file but the PR contain also release related changes and a change in the product licensing

File: CHANGELOG.md (hunk 1 above)
Comment (izgeri) [reply CHANGELOG.md:13]: I'm not sure about the history of this change log file, but across our suite this is common.

File: Jenkinsfile (hunk 2 above)
Comment (izgeri) [reply Jenkinsfile:112]: there is a separate commit titled "only publish on tags". I think that takes care of this concern

File: conjur-cli.gemspec (hunk 3 above)
Comment (izgeri) [reply conjur-cli.gemspec:7]: :)

File: conjur-cli.gemspec (hunk 4 above)
Comment (izgeri) [reply conjur-cli.gemspec:10]: the licensing actually didn't change - it's been Apache 2.0 for a long time, we just had it wrong in the gemspec :(

**Review by Tovli — APPROVED** [review]
Delightful as always
Just got a bit smarter by this PR

**Pull request merged**